microprofile-config/src/test/java/org/wildfly/quickstarts/AbstractRuntimeIT.java

```diff
@@ -0,0 +1,94 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.quickstarts;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * The foundation for runtime integration tests.
+ * @author emartins
+ */
+public abstract class AbstractRuntimeIT {
+
+    private final String serverHost;
+
+    /**
+     *
+     * @param defaultRootPath the root path that should be used if not overridden by properties
+     */
+    public AbstractRuntimeIT(String defaultRootPath) {
+        String serverHost = System.getenv("SERVER_HOST");
+        if (serverHost == null) {
+            serverHost = System.getProperty("server.host");
+        }
+        if (serverHost == null) {
+            serverHost = "http://localhost:8080/"+defaultRootPath;
+        }
+        this.serverHost = serverHost;
+    }
+
+    /**
+     * Retrieves the application's HTTP endpoint for the specified relativePath.
+     * @param relativePath the relative path
+     * @return an URI pointing to 'http://localhost:8080/defaultRootPath/relativePath' if the 'server.host' property is not set, 'server.host/relativePath' otherwise.
+     */
+    protected URI getHTTPEndpoint(String relativePath) {
+        try {
+            return new URI(serverHost + relativePath);
+        } catch (URISyntaxException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    /**
+     * Retrieves the application's HTTP endpoint with no relativePath.
+     * @return an URI pointing to 'http://localhost:8080/defaultRootPath' if the 'server.url' property is not set, 'server.host' otherwise.
+     */
+    protected URI getHTTPEndpoint() {
+        try {
+            return new URI(serverHost);
+        } catch (URISyntaxException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    /**
+     * Tests if the quickstart's HTTP endpoint is available.
+     * @param relativePath the web page's relative path
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    public void testHTTPEndpointIsAvailable(String relativePath) throws IOException, InterruptedException {
+        final HttpClient client = HttpClient.newBuilder()
+                .followRedirects(HttpClient.Redirect.ALWAYS)
+                .connectTimeout(Duration.ofMinutes(1))
+                .build();
+        final HttpRequest request = HttpRequest.newBuilder()
+                .uri(getHTTPEndpoint(relativePath))
+                .GET()
+                .build();
+        final HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+        assertEquals(200, response.statusCode());
+    }
+}
```

Comment:
This is pretty much the thing I have thought but in the end could not commit myself too go for it... I know this will add a bit of weird maintenance, after all we are talking about 49 quickstarts, but:
* each quickstart should be a fully independent project, we should include in its module all the user app needs
* it's really just one class, it would be weird to go for an SPI only containing it
* I can't imagine we changing it much

Honestly I think I would prefer to merge the abstract logic on the tests classes, than creating a test SPI